exporter/mezmoexporter/exporter.go

```diff
@@ -0,0 +1,174 @@
+	Meta      map[string]string `json:"meta"`
+}
+
+type MezmoLogBody struct {
+	Lines []MezmoLogLine `json:"lines"`
+}
+
+func newLogsExporter(config *Config, settings component.TelemetrySettings) *mezmoExporter {
+	var e = &mezmoExporter{
+		config:   config,
+		settings: settings,
+	}
+	return e
+}
+
+func (m *mezmoExporter) pushLogData(ctx context.Context, ld plog.Logs) error {
+	m.wg.Add(1)
+	defer m.wg.Done()
+
+	return m.logDataToMezmo(ld)
+}
+
+func (m *mezmoExporter) start(_ context.Context, host component.Host) (err error) {
+	client, err := m.config.HTTPClientSettings.ToClient(host.GetExtensions(), m.settings)
+	if err != nil {
+		return err
+	}
+
+	m.client = client
+
+	return nil
+}
+
+func (m *mezmoExporter) stop(context.Context) (err error) {
+	m.wg.Wait()
+	return nil
+}
+
+func (m *mezmoExporter) logDataToMezmo(ld plog.Logs) error {
+	var errs error
+
+	var lines []MezmoLogLine
+
+	// Convert the log resources to mezmo lines...
+	resourceLogs := ld.ResourceLogs()
+	for i := 0; i < resourceLogs.Len(); i++ {
+		ills := resourceLogs.At(i).ScopeLogs()
+		//resource := resourceLogs.At(i).Resource()
+		for j := 0; j < ills.Len(); j++ {
+			logs := ills.At(j).LogRecords()
+
+			for k := 0; k < logs.Len(); k++ {
+				var log = logs.At(k)
+
+				// Convert Attributes to meta fields being mindful of the maxMetaDataSize restriction
+				var attrs = map[string]string{}
+				attrs["trace.id"] = log.TraceID().HexString()
+				attrs["span.id"] = log.SpanID().HexString()
+				log.Attributes().Range(func(k string, v pcommon.Value) bool {
+					attrs[k] = truncateString(v.StringVal(), maxMetaDataSize)
+					return true
+				})
+
+				var s pcommon.Value
+				s, _ = log.Attributes().Get("appname")
+				var app = s.StringVal()
+
+				var line = MezmoLogLine{
+					Timestamp: log.Timestamp().AsTime().UTC().UnixMilli(),
+					Line:      truncateString(log.Body().StringVal(), maxMessageSize),
+					App:       truncateString(app, maxAppnameLen),
+					Level:     truncateString(log.SeverityText(), maxLogLevelLen),
+					Meta:      attrs,
+				}
+				lines = append(lines, line)
+			}
+		}
+	}
+
+	// Send them to Mezmo in batches < 10MB in size
+	var b strings.Builder
+	b.WriteString("{\"lines\": [")
+
+	var lineBytes []byte
+	for i, line := range lines {
+		if lineBytes, errs = json.Marshal(line); errs != nil {
+			return fmt.Errorf("error Creating JSON payload: %s", errs)
+		}
+
+		var newBufSize = b.Len() + len(lineBytes)
+		if newBufSize >= maxBodySize-2 {
+			str := b.String()
+			str = str[:len(str)-1] + "]}"
+			if errs = m.sendLinesToMezmo(str); errs != nil {
+				return errs
+			}
+			b.Reset()
+			b.WriteString("{\"lines\": [")
+		}
+
+		b.Write(lineBytes)
+
+		if i < len(lines)-1 {
+			b.WriteRune(',')
+		}
+	}
+
+	str := b.String() + "]}"
+	if errs = m.sendLinesToMezmo(str); errs != nil {
+		return errs
+	}
+
+	return nil
+}
+
+func (m *mezmoExporter) sendLinesToMezmo(post string) (errs error) {
+	var hostname = "otel"
```

Comment:
what is hostname meant to do?  "otel" seems like an odd value for it.  Should we be setting user-agent or something else instead?

Reply:
This is a temporary fix as `hostname` is a required field in the URI of the POST.  Once we remove this requirement from the backend, we will remove this as well.  The hostname attributes attached to each OTEL log entry are being passed along as metadata to our platform and are being utilized appropriately as the source that tagged the data intended.

Reply:
I'd suggest adding a TODO explaining why it is needed and when it can be removed.

If you wanted to, you could use the `host.name` resource attribute to fill this out.  But that's up to you.

Reply:
Added.